docs/docs/remark-ast.md

```diff
@@ -0,0 +1,270 @@
+var remark = require("remark")
+var visit = require("unist-util-visit")
+
+var tree = remark().parse("Some _emphasis_, **importance**, and `code`.")
+
+visit(tree, "text", visitor)
+
+function visitor(node) {
+  console.log(node)
+}
+```
+
+Here, it finds all text nodes and will `console.log` the nodes.
+
+Now with this, we can take the AST from our plugin and get working on adding functionality for headers:
+
+```js
+const visit = require("unist-util-visit")
+
+module.exports = ({ markdownAST }, pluginOptions) => {
+  visit(markdownAST, "heading", node => {
+    // Do stuff with heading nodes
+  })
+
+  return markdownAST
+}
+```
+
+Now we will visit all heading nodes and pass it into a transformer function which we can do whatever we wish.
+
+Looking again at the AST node for header:
+
+```JSON
+{
+  "type": "heading",
+  "depth": 1,
+  "children": [
+    {
+      "type": "text",
+      "value": "Hello World!",
+      "position": {
+        "start": { "line": 1, "column": 3, "offset": 2 },
+        "end": { "line": 1, "column": 15, "offset": 14 },
+        "indent": []
+      }
+    }
+  ],
+  "position": {
+    "start": { "line": 1, "column": 1, "offset": 0 },
+    "end": { "line": 1, "column": 15, "offset": 14 },
+    "indent": []
+  }
+},
+```
+
+We have context about the text as well as what depth the header is (for instance here we have a depth of 1 which would equate to an `h1` element)
+
+with the inner function of the `visit` call, I parse out all of the text and if it will map to a h1, I set the type of the node to `html` and set the node's value to be some custom html.
+
+```js
+const toString = require("mdast-util-to-string")
+
+visit(markdownAST, "heading", node => {
+  let { depth } = node
+
+  // Skip if not an h1
+  if (depth !== 1) return
+
+  // Grab the innerText of the heading node
+  let text = toString(node)
+
+  const html = `
+      <h1 style="color: rebeccapurple">
+        ${text}
+      </h1>
+    `
+
+  node.type = "html"
+  node.children = undefined
+  node.value = html
+})
+```
+
+A small library [mdast-util-to-string](https://github.com/syntax-tree/mdast-util-to-string) which is also made by Unified was used to extract the plaintext of the inner nodes, so this would remove links or other types of nodes inside the header, but given you have full access to the markdown AST, you can modify it however you wish.
+
+## Loading in changes and seeing effect.
+
+After doing such, we setup some pages to be programmatically created from markdown as shown in [Part 7 of the Gatsby Tutorial](/tutorial/part-seven/). Once this is set up, we can examine that our plugin works as seen below based on the markdown we wrote earlier.
+
+![Output](images/remark-ast-output.png)
+
+## Publishing the plugin
+
+To be shared with others, we can extract the plugin to its own directory outside of this site and then publish it to NPM so it can be accessed both on NPM and [Submitted to the Plugin Library](/contributing/submit-to-plugin-library).
```

Comment:
```suggestion
To be shared with others, you can extract the plugin to its own directory outside of this site and then publish it to NPM so it can be accessed both on NPM and [Submitted to the Plugin Library](/contributing/submit-to-plugin-library).
```